flax/core/scope.py

```diff
@@ -279,6 +290,9 @@ def reserve(self, name: str):
     Args:
       name: The name to reserve.
     """
+    if not isinstance(name, str):
+      raise ValueError('Variable and child scopes should have a string name.')
+    assert isinstance(name, str), 'v'
```

Comment:
I think you want to remove this redundant assert here?

Reply:
Done